### PR TITLE
fix: disable log by default in debug mode

### PR DESCRIPTION
**Pull request by AliAkberAakash**
## :scroll: Description
Reduce logging to console

## :bulb: Motivation and Context

Fix of issue #691 

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [ ] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
@AliAkberAakash thanks for doing this.
Please add an entry to the changelog and fix possible broken tests.

**Comment by AliAkberAakash** [comment]
I have added entry to change log and fixed broken tests. @marandaneto

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Unreleased
 
+* Fix: Disable log by default in debug mode
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion
* Fix: Disable log by default in debug mode (#753)
```

**Comment by marandaneto** [comment]
@AliAkberAakash 

>    info • Unused import: 'package:flutter/foundation.dart' • test/sentry_flutter_test.dart:1:8 • unused_import

**Comment by AliAkberAakash** [comment]
@marandaneto Hi, I have pushed the suggested changes, did u have a chance to look at it?

**Comment by marandaneto** [comment]
> @marandaneto Hi, I have pushed the suggested changes, did u have a chance to look at it?

Hi, There are still failing tests, see https://github.com/getsentry/sentry-dart/runs/5179416915?check_suite_focus=true
Is it related to your PR?

**Comment by brustolin** [comment]
My problem with this PR is that it changes a behavior already known to anyone using the SDK. And the solution for those who don't want logs is simple, just disable it in the options.

I would prefer to add a `debug` parameter to the SentryOptions constructor. Even though in flutter it is easy to initialize properties like this:

```dart
SentryOptions()..debug = false
```

**Comment by AliAkberAakash** [comment]
I'm not sure if this test is required anymore? or should this now check if the logger is noOpLogger right from the start? @marandaneto 
https://github.com/getsentry/sentry-dart/blob/a3d70071172e450a7afe5f21e4215493e02035e3/dart/test/sentry_test.dart#L320

**Comment by marandaneto** [comment]
> I'm not sure if this test is required anymore? or should this now check if the logger is noOpLogger right from the start? @marandaneto
> 
> https://github.com/getsentry/sentry-dart/blob/a3d70071172e450a7afe5f21e4215493e02035e3/dart/test/sentry_test.dart#L320

Since `debug` is false now, you have to add this `options.debug = true;` between L325 and L326

**Comment by marandaneto** [comment]
> My problem with this PR is that it changes a behavior already known to anyone using the SDK. And the solution for those who don't want logs is simple, just disable it in the options.
> 
> I would prefer to add a `debug` parameter to the SentryOptions constructor. Even though in flutter it is easy to initialize properties like this:
> 
> ```dart
> SentryOptions()..debug = false
> ```

Indeed, but that was the only SDK doing so, enabling `debug` by default, and it got too noisy, users can still `debug = true` manually, this won't affect production Apps but rather debugging only.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**